src/lib.rs

```diff
@@ -257,6 +257,12 @@ impl Type {
     /// Used for protocols such as UDP.
     pub const DGRAM: Type = Type(sys::SOCK_DGRAM);
 
+    #[cfg(target_os = "linux")]
```

Comment:
Can you also add a requirement on the `all` feature. This is used by socket to expose features that aren't available on all platforms, letting users know without workaround their application might not be cross-platform. See for example `SEQPACKET` just below (the `cfg_attr(docsrs, ..)` will also be needed.

Reply:
The two comment above also apply to all the other constants/functions.

Reply:
So all the functions and constants related to dccp should have this?
```rust
    #[cfg(all(feature = "all", target_os = "linux"))]
    #[cfg_attr(docsrs, doc(cfg(feature = "all", target_os = "linux")))]
```

Reply:
Yes